### PR TITLE
Add `--ignore-error` to reject

### PR DESCRIPTION
**Pull request by Tiggax**
# Description

Add `--ignore-errors` flag to reject.

This is a PR in reference to #10215 as select has the flag, but reject hasn't

# User-Facing Changes
user can now add `-i` or `--ignore-errors` flag to turn every cell path into option.

```nushell
> let arg = [0 5 a c]
> [[a b];[1 2] [3 4] [5 6]] | reject $a | to nuon
error index to large
# ----
> let arg = [0 5 a c]
> [[a b];[1 2] [3 4] [5 6]] | reject $a -i | to nuon
[[a, b]; [1, 2], [3, 4], [5, 6]]
```

### CODE REVIEW

**Comment by WindSoilder** [comment]
Hi, thanks for your contribution!
It seems that you want to ignore error if the given index is not exists.  Actually we can do something like this:
```nushell
[[a b];[1 2] [3 4] [5 6]] | reject 5?
```
Cell path which ends with `?` is optional cell path.  Which didn't raise error if it doesn't exists

**Comment by Tiggax** [comment]
@WindSoilder This PR was opened in reference to #10215 to standardize `reject`, `select` and `get` to have similar functionality.
I see what you mean, so i should have been more concise in the description. The `--ignore-errors` flag comes in handy when passing a list to the command. something like:
```nushell
let arg = [0 5 a c]
[[a b];[1 2] [3 4] [5 6]] | reject $a
```
where you cannot add `?` to make the `cell path` optional.
`select` solves this by including `--ignore-errors` flag.

Thinking now. It would make sense to make variables optional, as it would solve this issue, but right now this seems a simple solution that also expands to make the commands more similar.

**Comment by WindSoilder** [comment]
Ah, thanks for the clarification, I've checked and see relative pr to add `-i` back: #8488
Yeah it makes sence to me

File: crates/nu-command/tests/commands/reject.rs [hunk 1]
```diff
@@ -185,3 +185,9 @@ fn reject_multiple_rows_descending() {
     let actual = nu!("[[a,b];[1 2] [3 4] [5 6]] | reject 2 1 | to nuon");
     assert_eq!(actual.out, "[[a, b]; [1, 2]]");
 }
+
+#[test]
+fn test_ignore_errors_flag() {
+    let actual = nu!("[[a, b]; [1, 2], [3, 4], [5, 6]] | reject 5 -i | to nuon");
```
Comment (WindSoilder) [inline crates/nu-command/tests/commands/reject.rs:191]: So the main reason for `-i` is reject by variables, would you please provide a test for that case?

**Review by WindSoilder — CHANGES_REQUESTED** [review]

File: crates/nu-command/tests/commands/reject.rs (hunk 1 above)
Comment (Tiggax) [reply crates/nu-command/tests/commands/reject.rs:191]: Sure! I've added the test case for List of CellPaths :smile:

**Review by WindSoilder — APPROVED** [review]
Thanks!  LGTM, we'll make a release today, so it'll be hang on a bit.

**Comment by WindSoilder** [comment]
Thanks!

**Pull request merged**